spec/factories/csv_manifest.rb

```diff
@@ -5,15 +5,19 @@
 class CsvManifest
   include ActiveModel::Model
 
-  attr_accessor :locations, :labware_prefix, :number_of_labwares
+  attr_accessor :locations, :labware_prefix, :number_of_labwares, :positions
 
   def generate_csv
     n = 1
     CSV.generate do |csv|
-      csv << ['Box Barcode', 'Plate Barcode']
+      csv << ['Box Barcode', 'Plate Barcode', 'Position']
       locations.each do |location|
-        number_of_labwares.times do
-          csv << [location.barcode, "#{labware_prefix}#{pad_number(n)}"]
+        number_of_labwares.times do |num|
+          csv << if location.type == "UnorderedLocation"
+                   [location.barcode, "#{labware_prefix}#{pad_number(n)}"]
+                 else
+                   [location.barcode, "#{labware_prefix}#{pad_number(n)}", positions[num]].compact
+                 end
```

Comment:
I think this is probably fine but just to double check - does the page still work fine if you upload a csv with only two columns? i.e. the 'Position' header / column is not required? Just want to make sure it's backwards compatible and the users won't have to change what they're doing. Thanks

Reply:
Yep I tested this and it still works 👍  The position in the csv is just evaluated to nil if there is no defined position, and the procedure for adding the labware to the location is the same as before if this is the case.